qa/L0_shared_memory/shared_memory_test.py

```diff
@@ -29,98 +29,110 @@
 from tritonclient import utils
 
 
 class SharedMemoryTest(tu.TestResultCollector):
-    def test_invalid_create_shm(self):
-        # Raises error since tried to create invalid system shared memory region
-        try:
-            shm_op0_handle = shm.create_shared_memory_region(
-                "dummy_data", "/dummy_data", -1
-            )
-            shm.destroy_shared_memory_region(shm_op0_handle)
-        except Exception as ex:
-            self.assertTrue(str(ex) == "unable to initialize the size")
+    # Constant members
+    shared_memory_test_client_log = Path(os.getcwd()) / "client.log"
+    model_dir_path = Path(os.getcwd()) / "models"
+    model_source_path = Path(os.getcwd()).parents[0] / "python_models/add_sub/model.py"
+    model_config_source_path = (
+        Path(os.getcwd()).parents[0] / "python_models/add_sub/config.pbtxt"
+    )
 
-    def test_valid_create_set_register(self):
-        # Create a valid system shared memory region, fill data in it and register
-        if _protocol == "http":
-            triton_client = httpclient.InferenceServerClient(_url, verbose=True)
+    # Custom setup method to allow passing of parameters
+    def _setUp(self, protocol, log_file_path):
+        self._tritonserver_ipaddr = os.environ.get("TRITONSERVER_IPADDR", "localhost")
+        self._test_windows = bool(int(os.environ.get("TEST_WINDOWS", 0)))
+        self._timeout = os.environ.get("SERVER_TIMEOUT", 120)
+        self._protocol = protocol
+        self._test_passed = False
+        self._original_stdout, self._original_stderr = None, None
+        self._log_file_path = log_file_path
+
+        if self._protocol == "http":
+            self._url = self._tritonserver_ipaddr + ":8000"
+            self._triton_client = httpclient.InferenceServerClient(
+                self._url, verbose=True
+            )
         else:
-            triton_client = grpcclient.InferenceServerClient(_url, verbose=True)
-        shm_op0_handle = shm.create_shared_memory_region("dummy_data", "/dummy_data", 8)
-        shm.set_shared_memory_region(
-            shm_op0_handle, [np.array([1, 2], dtype=np.float32)]
+            self._url = self._tritonserver_ipaddr + ":8001"
+            self._triton_client = grpcclient.InferenceServerClient(
+                self._url, verbose=True
+            )
+        self._build_model_repo()
+        self._build_server_args()
+        self._shared_memory_test_server_log = open(log_file_path, "w")
+        self._server_process = util.run_server(
+            server_executable=self._server_executable,
+            launch_command=self._launch_command,
+            log_file=self._shared_memory_test_server_log,
+        )
+        util.wait_for_server_ready(
+            self._server_process, self._triton_client, self._timeout
+        )
+        # Redirect console output to client log and begin test
+        self._client_log = open(SharedMemoryTest.shared_memory_test_client_log, "a")
```

Comment:
I believe we can safely ignore these warnings. The file is opened during _setUp and closed in tearDown. According to the docs, tearDown will [always be called](https://docs.python.org/3/library/unittest.html#unittest.TestCase.tearDown:~:text=Method%20called%20immediately%20after%20the%20test%20method%20has%20been%20called%20and%20the%20result%20recorded.%20This%20is%20called%20even%20if%20the%20test%20method%20raised%20an%20exception) when setUp is successful. In this case, since we use a custom version of "setUp", the default unittest setUp method will always be successful, so the file should always be closed.